actors/builtin/multisig/multisig_actor.go

```diff
@@ -114,25 +105,31 @@ type TxnIDParams struct {
 func (a *MultiSigActor) Approve(rt vmr.Runtime, params *TxnIDParams) *vmr.EmptyReturn {
 	rt.ValidateImmediateCallerType(builtin.CallerTypesSignable...)
 	callerAddr := rt.ImmediateCaller()
-	a._rtValidateSignerOrAbort(rt, callerAddr)
-	a._rtApproveTransactionOrAbort(rt, params.ID)
+	var st MultiSigActorState
+	rt.State().Transaction(&st, func() interface{} {
+		a.validateAuthorizedParty(rt, &st, callerAddr)
+		return nil
+	})
+	a.approveTransaction(rt, params.ID)
```

Comment:
I don't like having two transaction to perform this operation. I suppose it was already like this, and your changes have just made it more clear. Can we move away from this pattern to a more performant one in a follow on?

Reply:
Yes, this PR makes it more clear that there are tons of unnecessary state load/store cycles which should be collapsed into many fewer! cc @zixuanzh  @jzimmerman